File: java/src/test/java/route/ImportTest.java
package route;

import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import java.io.FileNotFoundException;

import static org.junit.Assert.assertEquals;

public class ImportTest {

    private Driver driver;
    private JSONObject importJson = new JSONObject("{\n" +
            "  'name': 'Manual Import JAVA',\n" +
            "  'features': [{\n" +
            "    'type': 'segmentation',\n" +
            "    'segmentId': 14098,\n" +
            "    'emptyExisitingSegment': false\n" +
            "  }, {\n" +
            "    'type': 'duplicate',\n" +
            "    'rules': {\n" +
            "      'ignore': true\n" +
            "    }\n" +
            "  }, {\n" +
            "    'type': 'report',\n" +
            "    'sendFinalReport': true,\n" +
            "    'sendErrorReport': true,\n" +
            "    'contactGuids': ['044EAE5A'],\n" +
            "    'groupIds': []\n" +
            "  }, {\n" +
            "    'type': 'database',\n" +
            "    'updateExisting': true,\n" +
            "    'crushData': false\n" +
            "  }],\n" +
            "  'binding': 8330\n" +
            "}");
    private String id = "12978";

    @Before
    public void setUp() {
        JSONObject config = new JSONObject();

        config.put("xkey", "my_xkey");
        config.put("url", "http://127.0.0.1:8888/");
        driver = new Driver(config);
    }

    @Test
    public void testCreate() {
        JSONObject response = this.driver.imports.create(this.importJson);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testExecute() throws FileNotFoundException {
        JSONObject response = this.driver.imports.execute(id, "../assets/import.csv");
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testGet() {
        JSONObject response = this.driver.imports.get(this.id);
        assertEquals(response.getInt("statusCode"), 200);
    }

    @Test
    public void testList() {
        JSONObject response = this.driver.imports.list();
        assertEquals(response.getInt("statusCode"), 200);
    }

}

File: csharp/DriverTest/ImportTest.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Net;
using NP6Api;
using NP6Api.Models;
using RestSharp;

namespace DriverTest
{
    [TestClass]
    public class ImportTest
    {
        private static Driver _driver;
        private static ImportModel _import;
        private static JsonObject _importTest;

        [ClassInitialize()]
        public static void ImportTestInitialize(TestContext testContext)
        {
            var config = new Dictionary<string, string>();

            config["xKey"] = "YOUR XKEY";
            config["url"] = "http://127.0.0.1:8888/";
            _driver = new Driver(config);
            _import = new ImportModel()
            {
                Name = "Manual Import CSharp",
                Features = new Feature[]
                {
                    new Feature() {
                        Type = "segmentation",
                        SegmentId = 12345,
                        EmptyExistingSegment = false
                    },
                    new Feature() {
                        Type = "duplicate",
                        Rules = new Rule() { Ignore = true }
                    },
                    new Feature() {
                        Type = "report",
                        SendFinalReport = true,
                        SendErrorReport = true,
                        ContactGuids = new string[] {"0123ABC"},
                    },
                    new Feature() {
                        Type = "database",
                        UpdateExisting = true,
                        CrushData = false,
                    }
                },
                Binding = 1234
            };

            _importTest = new JsonObject
            {
                {"name", "Manual Import C#"},
                {"features", new JsonObject[] {

                    new JsonObject {
                        {"type", "segmentation"},
                        {"segmentId", 12345},
                        {"emptyExistingSegment", false}
                    },
                    new JsonObject {
                        {"type", "duplicate"},
                        {"rules", new JsonObject {
                             {"ignore", true}
                                    }
                        }
                    },
                    new JsonObject {
                        {"type", "report"},
                        {"sendFinalReport", true},
                        {"sendErrorReport", true},
                        {"contactGuids", new string[] {"01234ABC"} },
                        {"groupIds", new string[0] }
                    },
                    new JsonObject {
                        {"type", "database"},
                        {"updateExisting", true},
                        {"crushData", false},
                    }
                    }
                },
                {"binding", 1234}
            };
        }

        [TestMethod]
        public void TestCreate()
        {
            var response = _driver.Import.Create(_import);

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestExecute()
        {
            var response = _driver.Import.Execute("12345", "./import.csv");

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestGet()
        {
            var response = _driver.Import.Get("12345");

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestList()
        {
            var response = _driver.Import.List();

            Assert.AreEqual((HttpStatusCode)200, response.StatusCode);
        }

        [TestMethod]
        public void TestDelete()
        {
            var response = _driver.Import.Delete("12345");

            Assert.AreEqual((HttpStatusCode)204, response.StatusCode);
        }
    }
}
